App/Sagas/TextileSagas.ts

```diff
@@ -93,26 +94,43 @@ export function * updateNodeOverview ( action: ActionType<typeof TextileNodeActi
   }
 }
 
+export function * onboardedSuccess(action: ActionType<typeof PreferencesActions.onboardedSuccess>) {
+  let defaultThread: TT.Thread = yield call(getDefaultThread)
+  if (!defaultThread) {
+    // Skipping the normal addThread request here because we don't
+    // want to redirect to the default thread after complete
+    const thread: Thread = yield call(TextileNode.addThread, 'default' as TT.ThreadName)
+    yield put(ThreadsActions.addThreadSuccess(thread))
+    yield put(ThreadsActions.refreshThreadsRequest())
+  }
+}
+
 export function * handleProfilePhotoSelected(action: ActionType<typeof UIActions.selectProfilePicture>) {
   yield put(PreferencesActions.onboardedSuccess())
   yield call(NavigationService.navigate, 'PrimaryNavigation')
-  yield take(getType(TextileNodeActions.startNodeSuccess))
 
-  let defaultThread: TT.Thread | undefined = yield call(getDefaultThread)
-  if (!defaultThread) {
-    yield put(ThreadsActions.addThreadRequest('default' as TT.ThreadName))
-    const action: ActionType<typeof ThreadsActions.addThreadSuccess> = yield take(getType(ThreadsActions.addThreadSuccess))
-    defaultThread = action.payload.thread
-    yield put(ThreadsActions.refreshThreadsRequest())
+  // PreferencesActions.onboardedSuccess will setup the node/default thread, so wait for those
+  let online = yield select(TextileNodeSelectors.online)
+  let defaultThread: TT.Thread = yield call(getDefaultThread)
+  while (!online && !defaultThread) {
+    yield call(delay, 150)
+    online = yield select(TextileNodeSelectors.online)
+    defaultThread = yield call(getDefaultThread)
   }
   yield * processAvatarImage(action.payload.uri, defaultThread)
 }
 
 export function * handleProfilePhotoUpdated(action: ActionType<typeof UIActions.updateProfilePicture>) {
   yield call(NavigationService.navigate, 'TabNavigator')
 
-  let defaultThread: TT.Thread = yield call(getDefaultThread)
+  // Ensure that our node is online before we try to update the profile
+  let online = yield select(TextileNodeSelectors.online)
+  while (!online) {
+    yield call(delay, 150)
+    online = yield select(TextileNodeSelectors.online)
+  }
 
+  let defaultThread: TT.Thread = yield call(getDefaultThread)
```

Comment:
This logic is a little different than above, but same comment applies.

Reply:
okay, was mostly piling on the safety checks last night racking my brain why it was all broken. removing the online part